src/peer_digest.cc

```diff
@@ -525,19 +508,16 @@ peerDigestFetchReply(void *data, char *buf, ssize_t size)
 }
 
 int
-peerDigestSwapInCBlock(void *data, char *buf, ssize_t size)
+peerDigestSwapInCBlock(DigestFetchState * const fetch, const char *buf, const ssize_t size)
 {
-    DigestFetchState *fetch = (DigestFetchState *)data;
-
     assert(fetch->state == DIGEST_READ_CBLOCK);
 
-    if (peerDigestFetchedEnough(fetch, buf, size, "peerDigestSwapInCBlock"))
+    if (peerDigestFetchedEnough(fetch, size, "peerDigestSwapInCBlock"))
         return -1;
 
     if (size >= (ssize_t)StoreDigestCBlockSize) {
-        PeerDigest *pd = fetch->pd;
+        const auto pd = fetch->pd.get();
 
-        assert(pd);
```

Comment:
I hope we can restore this assertion because peerDigestFetchedEnough() above guarantees that the PeerDigest object is still valid.

Reply:
Ok, let's restore it. However I thought that we decided minimize these 'assert(pd)' throughout the code, relying on peerDigestFetchedEnough() guarantees of existing pd.